File: cpp/leetcode/338.cpp
class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> res(num + 1, 0);
        for (int i = 1; i <= num; ++i) {
            res[i] = res[i & (i - 1)] + 1;
        }
        return res;
    }
};

class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> res{0};
        for (int i = 1; i <= num; ++i) {
            if (i % 2 == 0) {
                res.push_back(res[i / 2]);
            } else {
                res.push_back(res[i / 2] + 1);
            }
        }
        return res;
    }
};

File: python/leetcode/338.py
class Solution(object):
    def countBits(self, num):
        ans = [0]
        for x in range(1, num + 1):
            ans += ans[x & (x - 1)] + 1,
        return ans


class Solution(object):
    def countBits(self, num):
        bits = [0] * (num + 1)
        for i in xrange(num + 1):
            bits[i] = bits[i / 2]
            if i % 2 != 0:
                bits[i] += 1
        return bits
